features/steps/shopcart_steps.py

```diff
@@ -90,3 +90,22 @@ def step_impl(context, product_id):
 def step_impl(context, user_id):
     element = context.driver.find_element_by_id('shopcarts-table-list')
     assert not element.find_elements_by_id('shopcart-' + user_id + '-row')
+
+@then(u'I should see "{message}" on the cart page')
+def step_impl(context, message):
+    parent = context.driver.find_element_by_class_name('mb-3')
+    child = parent.find_element_by_class_name('card-body')
+    assert message in child.text
+
+@when(u'I add "{quantity}" of Product "{product_id}" to the cart')
+def step_impl(context, quantity, product_id):
+    assert context.driver.find_element_by_id('product-' + product_id + '-quantity' )
```

Comment:
Why are you asserting the element exists before using it? Either way the test will fail. Same on line 109 and 75.

Reply:
You take this @radhikamattoo. It does not fail. why should it be a problem to assert before using ?

Reply:
Line 109 is checking that the row for the newly added product actually exists. It would throw an error otherwise.

I didn't implement the changes in lines 102 and 75, but line 75's is redundant since line 74 would throw an error if `btn` didn't exist. I honestly don't know how line 102 is passing lol. I'll take a look at it now

Reply:
Oh I did 75 just because I was trying to understand where the problem was. It can go.

I now see that I did 102, is passing because that element always exists. The reason was that I was checking If I was coding correctly in VUE because my chrome was showing me cached versions of the app whenever I coded something wrong and that line caught the errors. So feel free to remove it. @radhikamattoo